packages/network-controller/tests/fake-provider-engine.ts

```diff
@@ -117,10 +119,70 @@ export class FakeProviderEngine extends EventEmitter implements ProviderEngine {
     this.#isStopped = true;
   }
 
-  sendAsync<P, V>(
+  send = async <P>(payload: SendAsyncPayload<P>) => {
```

Comment:
Is there any way to reuse the logic in `sendAync` for `send`? This logic is a bit complex and I feel like repeating it makes it a bit more overwhelming.